std/machines/small_field/rotate.asm

```diff
@@ -0,0 +1,71 @@
+use std::utils::unchanged_until;
+use std::utils::cross_product;
+use std::convert::int;
+
+// Rotate for single bytes using an exhaustive table, returning two 16-bit values as result.
+// TODO this way, we cannot prove anything that rotates by more than 31 bits.
+machine ByteRotate with
+    latch: latch,
+    operation_id: operation_id,
+    degree: 65536
+{
+    // P_C0 and P_C1 are both 16 bit limbs of P_C, where P_C0 is the less significant limb.
+    operation run<0> P_operation, P_A, P_B, P_ROW -> P_C0, P_C1;
+
+    col fixed latch = [1]*;
+    col fixed operation_id = [0]*;
+
+    let bit_counts = [256, 32, 4, 2];
+    let min_degree = std::array::product(bit_counts);
+    std::check::assert(std::prover::min_degree() >= std::array::product(bit_counts), || "The rotate machine needs at least 65536 rows to work.");
+    std::check::assert(std::field::modulus() >= 65535, || "The field modulo should be at least 2^16 - 1 to work in the rotate machine.");
+    let inputs = cross_product(bit_counts);
+    let a: int -> int = inputs[0];
+    let b: int -> int = inputs[1];
+    let row: int -> int = inputs[2];
+    let op: int -> int = inputs[3];
+    let P_A: col = a;
+    let P_B: col = b;
+    let P_ROW: col = row;
+    let P_operation: col = op;
+    let c: int -> int = |i| match op(i) {
```

Comment:
Can we extract his to a "common" file?

Reply:
not sure what's best idea to do it

Reply:
Isn't it better to refactor this later? Cause this one also includes in the shift machine. I think this can be in a different PR

Reply:
The problem is we have both the same generic for the 1. different fields and the 2. different machines. For poseidon, we need to share the same code across the "different fields" and for shift and rotate, we can share all the  code both "different fields" and "different machines". So, if we put the generic.asm inside the machine folder for now, it would be better?

Reply:
https://github.com/powdr-labs/powdr/issues/1903 Opened a new issue for it

Reply:
Cause, I need to test the copy-pasted code for the bigger circuit without getting rid of productivity, but I need to be conscious that I'm adding a new code which needs to be refactor with another circuit

Reply:
@chriseth Should we do it now or in a different PR?

Reply:
@chriseth What to do next?

Reply:
If you actually do it, then it's fine.